deepchecks/vision/checks/distribution/train_test_label_drift.py

```diff
@@ -0,0 +1,447 @@
+from deepchecks.vision import VisionDataset
+import numpy as np
+from collections import Counter
+import plotly.graph_objs as go
+
+
+__all__ = ['TrainTestLabelDrift']
+
+
+class TrainTestLabelDrift(TrainTestBaseCheck):
+    """
+    Calculate label drift between train dataset and test dataset, using statistical measures.
+
+    Check calculates a drift score for the label in test dataset, by comparing its distribution to the train
+    dataset.
+    For numerical columns, we use the Earth Movers Distance.
+    See https://www.lexjansen.com/wuss/2017/47_Final_Paper_PDF.pdf
+    For categorical columns, we use the Population Stability Index (PSI).
+    See https://en.wikipedia.org/wiki/Wasserstein_metric.
+
+
+    Args:
+        max_num_categories (int):
+            Only for categorical columns. Max number of allowed categories. If there are more,
+            they are binned into an "Other" category. If max_num_categories=None, there is no limit. This limit applies
+            for both drift calculation and for distribution plots.
+    """
+
+    def __init__(
+            self,
+            max_num_categories: int = 10
+    ):
+        super().__init__()
+        self.max_num_categories = max_num_categories
+
+    def run(self, train_dataset, test_dataset, model=None) -> CheckResult:
+        """Run check.
+
+        Args:
+            train_dataset (VisionDataset): The training dataset object.
+            test_dataset (VisionDataset): The test dataset object.
+            model: not used in this check.
+
+        Returns:
+            CheckResult:
+                value: dictionary of column name to drift score.
+                display: distribution graph for each column, comparing the train and test distributions.
+
+        Raises:
+            DeepchecksValueError: If the object is not a Dataset or DataFrame instance
+        """
+        return self._calc_drift(train_dataset, test_dataset)
+
+    def _calc_drift(
+            self,
+            train_dataset: VisionDataset,
+            test_dataset: VisionDataset,
+    ) -> CheckResult:
+        """
+        Calculate drift for all columns.
+
+        Args:
+            train_dataset (VisionDataset): The training dataset object. Must contain a label.
+            test_dataset (VisionDataset): The test dataset object. Must contain a label.
+
+        Returns:
+            CheckResult:
+                value: drift score.
+                display: label distribution graph, comparing the train and test distributions.
+        """
+        train_dataset = VisionDataset.validate_dataset(train_dataset)
+        test_dataset = VisionDataset.validate_dataset(test_dataset)
+        train_dataset.validate_label()
+        test_dataset.validate_label()
+
+        task_type = train_dataset.label_type
+        displays = []
+
+        if task_type == TaskType.CLASSIFICATION.value:
+
+            train_label_distribution = train_dataset.get_samples_per_class()
```

Comment:
not really related to your PR, but think this should be stored as it may be costly (currently it's done per call)